### PR TITLE
Switched JSHint for ESLint with SparkPost config

### LINKED ISSUE #155: Switch to using ESLINT
LINT the library using ESLINT. Use a npm script.

### PR DESCRIPTION
**Pull request by aydrian**
Removed JSHint
Added npm lint script
Switch Grunt to call `npm run lint` until Grunt is removed
Resolved all new linting errors
Closes #155

### CODE REVIEW

**Comment by aydrian** [comment]
Currently not linting examples directory.

File: lib/SendGridCompatibility/Email.js [hunk 1]
```diff
@@ -5,8 +5,9 @@
  *
  * @param options object that contains initial values
  */
-function Email(options){
-  for (var option in options) {
```
Comment (orval) [inline lib/SendGridCompatibility/Email.js]: An alternative is to use `for (let option in options) {`.

Comment (aydrian) [inline lib/SendGridCompatibility/Email.js]: Truthfully, all that SendGridCompatibility is going away. But I'll remember that. Now that we are only supporting node 4+, we can do all the ES6 stuff.

**Comment by orval** [comment]
There may be cases where `let` can be used instead of `var` to save moving them to the top of the function. Probably doesn't matter either way. Otherwise lgtm.

**Comment by aydrian** [comment]
I just wanted it to pass. I'm going to have to go through and refactor each library so I might do that.

File: lib/SendGridCompatibility/index.js [hunk 2]
```diff
@@ -13,14 +14,15 @@ var SparkPost = require('../sparkpost');
  * @param apiKey: api key string
  * @param options: optional additional options object
  */
-var sendgrid = function(username, apiKey, options) {
+sendgrid = function(username, apiKey, options) {
+  var urlOpts, opts;
```
Comment (colestrode) [inline lib/SendGridCompatibility/index.js]: usually we split vars on separate lines. this is fine for now, that would make a lot of updates for you for basically no meaningful impact :)

**Pull request merged**